src/c++/perf_analyzer/docs/data_guide.md

```diff
@@ -0,0 +1,291 @@
+for a model with single input "INPUT", shape [-1,-1] and data type INT32:
+
+```
+  {
+    "data" :
+     [
+        {
+          "INPUT" :
+                {
+                    "content": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    "shape": [2,8]
+                }
+        },
+        {
+          "INPUT" :
+                {
+                    "content": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    "shape": [8,2]
+                }
+        },
+        {
+          "INPUT" :
+                {
+                    "content": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+                }
+        },
+        {
+          "INPUT" :
+                {
+                    "content": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    "shape": [4,4]
+                }
+        }
+        ...
+      ]
+  }
+```
+
+The following is the example to provide contents as base64 string with explicit shapes:
+
+```
+{
+  "data": [{
+      "INPUT": {
+                 "content": {"b64": "/9j/4AAQSkZ(...)"},
+                 "shape": [7964]
+               }},
+    (...)]
+}
+```
+
+Note that for STRING type an element is represented by a 4-byte unsigned integer giving
+the length followed by the actual bytes. The byte array to be encoded using base64 must
+include the 4-byte unsigned integers.
+
+### Output Validation
+
+When real input data is provided, it is optional to request perf analyzer to
+validate the inference output for the input data.
+
+Validation output can be specified in "validation_data" field in the same format
+as "data" field for real input. Note that the entries in "validation_data" must
+align with "data" for proper mapping. The following example describes validation
+data for a model with inputs named, INPUT0 and INPUT1, outputs named, OUTPUT0
+and OUTPUT1, all tensors have shape [4, 4] and data type INT32:
+
+```
+  {
+    "data" :
+     [
+        {
+          "INPUT0" : [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+          "INPUT1" : [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+        }
+        ...
+      ],
+    "validation_data" :
+     [
+        {
+          "OUTPUT0" : [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+          "OUTPUT1" : [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
+        }
+        ...
+      ]
+  }
+```
+
+Besides the above example, the validation outputs can be specified in the same
+variations described in "real input data" section.
+
+## Shared Memory
+
+By default perf_analyzer sends input tensor data and receives output
+tensor data over the network. You can instead instruct perf_analyzer to
+use system shared memory or CUDA shared memory to communicate tensor
+data. By using these options you can model the performance that you
+can achieve by using shared memory in your application. Use
+[--shared-memory=system](cli.md#--shared-memory=[none|system|cuda])
```

Comment:
This might be browser-specific (hopefully not), but the link here seems wrong. Here's what my browser says is the correct heading:

`cli.md#--shared-memorynonesystemcuda`

Reply:
all links should be updated

Reply:
as part of this epic, we should create a L0_pa_doc_links test similar to what is in MA and triton server repos. It will flag if any of these are incorrect